### ISSUE TITLE
n00b observations / questions :)

### ISSUE THREAD

**Issue by kevinctofel** [scripting/publicfolder#4]
Let me preface this by saying, I have **_never_** used Node.js. I was able to fiddle around to get it installed and now have PublicFolder running. I'm really close to having it work because I do have it reading my local watchFolder:

![Screenshot from 2020-05-23 11-12-24](https://user-images.githubusercontent.com/16373212/82734122-a1d93800-9ce6-11ea-99c0-5c38ebe025fc.png)

One thing that I didn't realize from the documentation is that aside from having Node.js installed, you also have to install the AWS software. When I read "the AWS software, which is part of publicFolder" I mistook "part of publicFolder" to mean it's already in there. ;) 

Anyway, it was a quick "apt install" for me on Linux once I realized this.

My only issue now (I think) at this point is app/user access to my S3 bucket. I did create a user and assigned it S3 permissions. Got my access key and secret key. I then used the "aws configure" command to set these which created a configuration file in .aws, so all good on that. But getting access denied when running helloworld.js with node: 

![Screenshot from 2020-05-23 11-10-48](https://user-images.githubusercontent.com/16373212/82734228-96d2d780-9ce7-11ea-9167-942e64ada938.png)

I totally get that this is an AWS permissions issue I'm running into, which is beyond the scope of publicFolder; I'll keep tinkering in the AWS web console because I think that's where I have to fix this or attach this user to my bucket somehow. Cheers!

**Comment by scripting** [comment]
Kevin thanks for doing this. It's very helpful.

But back up a couple of steps -- you do _not_ need to install the AWS software. So everything you've been trying to do that way has nothing to do with getting publicFolder to work.

**Comment by scripting** [comment]
So starting at the top -- I don't know what the first screen shot is a picture of. Obviously a list of files, but I don't think it's something that my software put out. 

My guess is that you need to create the credentials text file in ~/.aws/credentials

That's the tricky bit. Let me see if I point to the AWS instructions for that.

**Comment by scripting** [comment]
OK this is what a credentials files looks like.

https://gist.github.com/scripting/16dbcfdd4ec2648d6935d7413dbcceca

I do point to [the docs](https://aws.amazon.com/blogs/security/a-new-and-standardized-way-to-manage-credentials-in-the-aws-sdks/), but like all AWS docs, it's a slog to get through them. But your goal is to get a credentials file at this location on your local computer with the proper values for your account.

` ~/.aws/credentials`

**Comment by kevinctofel** [comment]
Gotcha! Just uninstalled the AWS CLI although I do like the "aws configure" command that lets you set your creds for the text file. I actually do have a similar-looking credentials.txt file in a hidden aws directory. Obviously my keys are different. ;) Got them from the AWS Console where I set up an IAM user with S3 permissions. It's possible I've done something incorrect there so I'll double-check. I don't see a way to add user access to my S3 bucket on the S3 page in the Amazon console so that's why I went the IAM route...

BTW: That first screenshot above is in a subfolder called "stats" that was created when I ran the app. I figured it's a log of what files are in my watchFolder because that's what it shows. 

One thing I noticed in the S3 permissions I was trying to set up for my user: I don't see a ListObjects action and that's what the error message is saying I don't have access to, as shown in the second screenshot. Does that help from troubleshooting standpoint?

**Comment by scripting** [comment]
Be more careful about saying 1. what you did, 2. what happened and 3. what you expected to happen. A lot of what you're saying here and earlier is leaving me with little idea what happened.

The ~/.aws/credentials file _is_ in a hidden folder, so maybe you already have one. Your job right now is to get that working. The location on S3 has to be public, btw -- the docs are wrong in saying it can be private. (It can, but let's get to that later.)

Also please share your config.json file here. I think there might be a mistake in there too.

And SLOW DOWN and carefully explain what you're doing. Very often you'll spot the problem that way without your needing help.

**Comment by kevinctofel** [comment]
Sorry for being a little all over the place. I did have the S3 location set to private until now. I've changed that to public on the AWS Console.

Here is my config.json file contents:

{
	"watchFolder": "/home/kevin/Code/GitHubBackup/",
	"s3Folder": "s3.us-east-1.amazonaws.com/kctofelpublicfoldernodejs/",
	"urlS3Folder": ""
	}

Now that I'm making this folder public, the instructions tell me to put the URL of the S3 folder in the above configuration. 

I'm going to wait on that until you read this because I want to see if you think the S3 bucket configuration string above looks to be in the correct format or not.

**Comment by kevinctofel** [comment]
Actually, I now see what my issue was: The s3Folder in the config file. I've changed it to "s3Folder": "/kctofelpublicfoldernodejs/" and the app is working as expected. The sub-folders and their contents that are in my watchFolder appear on S3! Thanks for your patience! 

![Screenshot from 2020-05-23 13-47-01](https://user-images.githubusercontent.com/16373212/82737050-08b51c00-9cfc-11ea-88b9-130671c40f6c.png)

**Comment by scripting** [comment]
Very good! Sometimes you have to just stare at things for a while and that gives you ideas for things to try. 

So here's another idea -- can you figure out how to turn your public bucket into a website? You can do it all through the S3 user interface. And then you can publish a blog or whatever you want there using tools on your desktop.

That's where the real power of this stuff comes in.

Also if you want to make it private, there's a note on the most recent release at the end of the readme file for publicFolder that says what to do.

**Comment by kevinctofel** [comment]
Indeed. I will say I got a little confused by nomenclature due to my n00b-iness. What you call a "folder", AWS calls a "bucket". With that in my head, I created my bucket and then created a folder inside of it, which I called "s3Folder". And away I went down the wrong path. ;)

Funny you should mention turning it into a public site. I've been reading a bunch lately on Static Site Generators and was considering a personal blog hosted on Google Firebase as it's basically free. And I really need a personal site for all of the "Not About Chromebook" things that don't belong on AboutChromebooks.com. ;) 

I think I'll draft up some basic HTML & CSS locally and keep those in my watchFolder so I can use S3 and publicFolder to host the content online there. Learned a bunch today, so thanks! :)
